lib/Service/VerdictService.php

```diff
@@ -252,4 +245,27 @@ private function createAndConnectVaas(): Vaas {
 		$vaas->Connect($this->authenticator->getToken());
 		return $vaas;
 	}
+
+	/**
+	 * Checks if the file is in the blocklist or not in the allowlist and throws an exception if it is not allowed to scan the file.
+	 * @param string $filePath
+	 * @return bool
+	 */
+	public function isAllowedByAllowAndBlocklist(string $filePath): bool {
+		$blocklist = $this->getBlocklist();
+		$this->logger->debug("Blocklist: " . implode(", ", $blocklist));
+		foreach ($blocklist as $blocklistItem) {
+			if (str_contains(strtolower($filePath), strtolower($blocklistItem))) {
+				return false;
+			}
+		}
+		$allowlist = $this->getAllowlist();
+		$this->logger->debug("Allowlist: " . implode(", ", $allowlist));
+		foreach ($allowlist as $allowlistItem) {
```

Comment:
Maybe I just don't get it, or the semantic is intended like that,... but doesn't the code here mean that a filename has to match _all_ entries in the allow list, instead of one? Say I want to allow you to scan only "eicar.txt" and "eicar.com" and I put both into the allow list. Then as a result neither would be scanned, because a file cannot match both.
(In this case I could make one entry "eicar." as a kind of wildcard, but the behavior would be very surprsing to me..)